semantic-analysis/src/Analysis/Name.hs

```diff
@@ -1,40 +1,67 @@
-{-# LANGUAGE DeriveTraversable, GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
 module Analysis.Name
-( Name(..)
-, Named(..)
-, named
-, named'
-, namedName
-, namedValue
+( Name (Name)
```

Comment:
I’ve chosen to only export the `Name` constructor for now, but I’m considering making it 100% hidden and building

```
eliminate :: (Text -> a) -> (Int -> a) -> (Name -> a)
```

in case we need to pattern match.